src/core/dfn.js

```diff
@@ -238,3 +243,37 @@ function addContractDefaults() {
     dfn.dataset.export = "";
   }
 }
+
+// - Sets data-defines on well-known definition content patterns
+function addDefinitionPointers() {
+  // A dt with class definition (or in a dl with such a class)
+  // containing a definition
+  // indicates that the following dd or div element contains its prose content
+  /** @type NodeListOf<HTMLElement> */
+  const describedDTs = document.querySelectorAll(
+    "dl.definition dt:has(dfn[data-dfn-type]), dt.definition:has(dfn[data-dfn-type])"
+  );
+  for (const dt of describedDTs) {
+    const dfnId = dt.querySelector("dfn[data-dfn-type]").id;
+
+    const dfnContent = /** @type {HTMLElement | null} */ (
+      dt.nextElementSibling
+    );
+    if (dfnContent && dfnId && ["DIV", "DD"].includes(dfnContent.tagName)) {
+      dfnContent.dataset.defines = `#${dfnId}`;
+    }
+  }
+
+  // a non-dt element with class definition containing a definition
+  // indicates that the said element contains its prose content
+  /** @type NodeListOf<HTMLElement> */
+  const otherDescriptionContainers = document.querySelectorAll(
+    ":not(dt):not(dl).definition:has(dfn[data-dfn-type])"
```

Comment:
Nit: I wonder if we should just iterate over `.definition:has(dfn[data-dfn-type])` and do the check/processing in a single loop. Keeps selectors simpler, and avoids some duplication (the `dataset.defines = #{dfnId}` part).

Reply:
the new approach solves this as well